### PR TITLE
saves the correct subwindow size when it is hidden

### PR DESCRIPTION
**Pull request by BaraMGB**
fixes https://github.com/LMMS/lmms/issues/3584

### CODE REVIEW

**Comment by Wallacoloo** [comment]
Now that the `sizeIfInvisible` parameter is unused, it should probably be removed from the function signature as well.

**Comment by BaraMGB** [comment]
@Wallacoloo :+1:

**Comment by qnebra** [comment]
Possibly bug, how to reproduce:

1. Open lmms with this branch merged
2. New default project open automatically, like 
3. Try open Pianoroll, Automation editor, Project notes,
4. Nothing happens, they aren't shown or they just invisible. I don't know.
5. Try open Song Editor, B+B Editor, Mixer, Controller Rack
6. They opens like always will be.

GIF for proof:
http://imgur.com/fzemcyU

**Comment by BaraMGB** [comment]
@qnebra I can't reproduce. :(

**Comment by qnebra** [comment]
Ok, maybe it's fixed and I just making unnecesary noise.

File: include/MainWindow.h [hunk 1]
```diff
@@ -142,7 +142,7 @@ class MainWindow : public QMainWindow
 		return m_keyMods.m_alt;
 	}
 
-	static void saveWidgetState( QWidget * _w, QDomElement & _de, QSize const & sizeIfInvisible = QSize(0, 0) );
+	static void saveWidgetState(QWidget * _w, QDomElement & _de);
```
Comment (zonkmachine) [inline include/MainWindow.h]: Spaces at in the parenthesis.

**Comment by BaraMGB** [comment]
@qnebra has objections. I talked to him in discord. He can reproduce the problems after checkout this branch. There seems to be something fishy.

**Comment by qnebra** [comment]
I don't know exactly how it worked, in my case it look like "pianoroll", "automation editor", "project notes" windows dimensions are set to 0x0, so you can't find them, but they are active.

**Comment by zonkmachine** [comment]
@BaraMGB Can you rebase this against stable-1.2 ?
I added a default template in ac9f4a2 and if there is a bug with new projects/templates, this is going to affect this PR.

@qnebra 
Can you paste back the output of `git log --oneline -n 10` ?
Do you have a local default project template? (Save as default tempate?).
Can you try a clean build?

**Comment by qnebra** [comment]
@zonkmachine 

1. git log --oneline -n 10
805ed24 remove invisible size from saveWidgetState()
a82462f saves the correct subwindow size when it is hidden
f1a3368 Merge pull request #3583 from tresf/stable-1.2
745042d Fix CMP0050 compilation warnings
3607c92 Bitcrush Plugin Redesign (#3575)
38bca9a fix time signature updates timelinewidget also on pos 0 (#3572)
f8a508f Fixes #3563 (#3567)
6970c88 fix alternate shading for changing time signature (#3559)
c0b910e Replacement metronome samples (#3513)
300058d Flanger LFO rate syncronised (#3521)

2. Yes, I have.

3. I will try.

**Comment by zonkmachine** [comment]
> Ok guys, very, very sorry for noise. Create a new one default template fixed issue locally.

No, that is a concern. I still think we still need to test this with the new template fix.

**Comment by BaraMGB** [comment]
@zonkmachine done

**Comment by zonkmachine** [comment]
Works fine here like before.

@qnebra Can you build and test this? Test it both with and without the template you made.

**Comment by qnebra** [comment]
@zonkmachine Ok, I hope delete new default template will help test. Now install.

**Comment by zonkmachine** [comment]
That'll do. Merge! :tractor:

**Comment by BaraMGB** [comment]
:+1:  Thanks to @qnebra  and @zonkmachine !

**Comment by tresf** [comment]
Cherry-picked and ready for master in 220559f.

**Pull request merged**